src/Utils/CustomFieldRepoService.php

```diff
@@ -40,6 +45,63 @@ public function getEntitiesIdsForCustomFieldId($entityClass, $customFieldId)
         return array_column($result, 'id');
     }
 
+    /**
+     * @param string $customFieldId
+     * @param string $firstRootAlias
+     * @param \Doctrine\ORM\QueryBuilder $qb
+     *
+     * @return array id of entities, which fulfil any condition
+     */
+    protected function addAnyCustomFieldIdQueryResult($customFieldId, $firstRootAlias, $qb)
+    {
+        $qbCloned = clone $qb;
+        if (!in_array('cf', $qb->getAllAliases())) {
+            $qbCloned->join($firstRootAlias . '.customFields', 'cf');
+        }
+        $qbCloned->andWhere('cf.fieldId = :fieldId')
+            ->setParameter('fieldId', $customFieldId);
+
+        return $qbCloned->getQuery()->getResult();
+    }
+
+    /**
+     * Method looking for records, where any value of custom field is set.
+     *
+     * @param string $customFieldId
+     * @param string $firstRootAlias
+     * @param \Doctrine\ORM\QueryBuilder $qb
+     *
+     * @return \Doctrine\ORM\QueryBuilder
+     */
+    public function addAnyCustomFieldId($customFieldId, $firstRootAlias, $qb)
+    {
+        $entities = $this->addAnyCustomFieldIdQueryResult($customFieldId, $firstRootAlias, $qb);
+        $qb->andWhere($firstRootAlias . sprintf('.id IN (:any%s)', $this->parameterCount))
+            ->setParameter('any' . $this->parameterCount, $entities);
+        $this->parameterCount++;
+
+        return $qb;
+    }
+
+    /**
+     * Method looking for records, where no value of custom field is set.
+     *
+     * @param string $customFieldId
+     * @param string $firstRootAlias
+     * @param \Doctrine\ORM\QueryBuilder $qb
+     *
+     * @return \Doctrine\ORM\QueryBuilder
+     */
+    public function addNoneCustomFieldId($customFieldId, $firstRootAlias, $qb)
+    {
+        $entities = $this->addAnyCustomFieldIdQueryResult($customFieldId, $firstRootAlias, $qb);
+        $qb->andWhere($firstRootAlias . sprintf('.id NOT IN (:notAny%s)', $this->parameterCount))
```

Comment:
same question as above. Here we could make a left join and only allow the results where cf.id is NULL. We would need a separate join for all cf types (per $customFieldId).